components/function-controller/cmd/manager/configuration.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/kyma-project/kyma/components/function-controller/internal/controllers"
+	"github.com/kyma-project/kyma/components/function-controller/pkg/envars"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type configuration struct {
+	metricsAddr                   string
+	logDevModeEnabled             bool
+	leaderElectionEnabled         bool
+	leaderElectionNamespace       string
+	leaderElectionID              string
+	maxConcurrentReconciles       int
+	tektonRequestsCPU             resource.Quantity
+	tektonRequestsMem             resource.Quantity
+	tektonLimitsCPU               resource.Quantity
+	tektonLimitsMem               resource.Quantity
+	dockerRegistryPort            int
+	dockerRegistryFqdn            string
+	dockerRegistryExternalAddress string
+	dockerRegistryName            string
+	runtimeConfigmapName          string
+	imagePullSecretName           string
+	imagePullAccount              string
+	requeueDuration               time.Duration
+}
+
+func fnCtrlConf(appCfg *configuration) *controllers.FunctionControllerConfig {
+	return &controllers.FunctionControllerConfig{
+		Limits: &corev1.ResourceList{
+			corev1.ResourceCPU:    appCfg.tektonLimitsCPU,
+			corev1.ResourceMemory: appCfg.tektonLimitsMem,
+		},
+		Requests: &corev1.ResourceList{
+			corev1.ResourceCPU:    appCfg.tektonRequestsCPU,
+			corev1.ResourceMemory: appCfg.tektonRequestsMem,
+		},
+		MaxConcurrentReconciles:       appCfg.maxConcurrentReconciles,
+		DockerRegistryPort:            appCfg.dockerRegistryPort,
+		DockerRegistryFqdn:            appCfg.dockerRegistryFqdn,
+		DockerRegistryExternalAddress: appCfg.dockerRegistryExternalAddress,
+		DockerRegistryName:            appCfg.dockerRegistryName,
+		ImagePullSecretName:           appCfg.imagePullSecretName,
+		ImagePullAccount:              appCfg.imagePullAccount,
+		RuntimeConfigmap:              appCfg.runtimeConfigmapName,
+	}
+}
+
+type option = func() error
+
+// TODO LEADER_ELECTION_ID is not used atm - investigate
```

Comment:
I see it is used, is it some leftover comment or something is really not usable?

Reply:
yes, this is a leftover, thx